tests/by-util/test_sort.rs

```diff
@@ -1544,6 +1544,37 @@ fn test_files0_from_zero_length() {
         .stderr_only("sort: -:2: invalid zero-length file name\n");
 }
 
+#[test]
+#[cfg(target_os = "linux")]
+fn test_files0_from_non_utf8_filename() {
+    use std::ffi::OsStr;
+    use std::os::unix::ffi::OsStrExt;
+    use std::path::Path;
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    at.touch("file1.txt");
+    at.append("file1.txt", "zebra\n");
+    at.touch("file2.txt");
+    at.append("file2.txt", "apple\n");
+
+    // Create a file with non-UTF-8 bytes in filename
+    let non_utf8_name = OsStr::from_bytes(b"\xff\xfefile3.txt");
+    let non_utf8_path = at.plus_as_string("").to_owned() + "/";
+    let full_path = Path::new(&non_utf8_path).join(non_utf8_name);
```

Comment:
Unless I'm missing something, I think you can simplify it and use `plus`:
```suggestion
    let full_path = at.plus(non_utf8_name);
```